translate/cloud-client/src/test/java/com/example/translate/TranslateTextWithGlossaryAndModelIT.java

```diff
@@ -0,0 +1,136 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.translate;
+
+import static com.google.common.truth.Truth.assertThat;
+import static junit.framework.TestCase.assertNotNull;
+
+import com.google.api.gax.longrunning.OperationFuture;
+import com.google.cloud.translate.v3.CreateGlossaryMetadata;
+import com.google.cloud.translate.v3.CreateGlossaryRequest;
+import com.google.cloud.translate.v3.DeleteGlossaryMetadata;
+import com.google.cloud.translate.v3.DeleteGlossaryRequest;
+import com.google.cloud.translate.v3.DeleteGlossaryResponse;
+import com.google.cloud.translate.v3.GcsSource;
+import com.google.cloud.translate.v3.Glossary;
+import com.google.cloud.translate.v3.GlossaryInputConfig;
+import com.google.cloud.translate.v3.GlossaryName;
+import com.google.cloud.translate.v3.LocationName;
+import com.google.cloud.translate.v3.TranslationServiceClient;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Tests for Batch Translate Text With Glossary and Model sample. */
+@RunWith(JUnit4.class)
+@SuppressWarnings("checkstyle:abbreviationaswordinname")
+public class TranslateTextWithGlossaryAndModelIT {
+  private static final String PROJECT_ID = System.getenv("GOOGLE_CLOUD_PROJECT");
+  private static final String LOCATION = "us-central1";
+  private static final String MODEL_ID = "TRL2188848820815848149";
+  private static final String GLOSSARY_INPUT_URI =
+      "gs://cloud-samples-data/translation/glossary_ja.csv";
+  private static final String GLOSSARY_ID =
+      String.format("test_%s", UUID.randomUUID().toString().replace("-", "_").substring(0, 26));
+
+  private ByteArrayOutputStream bout;
+  private PrintStream out;
+
+  private static void requireEnvVar(String varName) {
+    assertNotNull(
+            System.getenv(varName),
+            "Environment variable '%s' is required to perform these tests.".format(varName)
+    );
+  }
+
+  @BeforeClass
+  public static void checkRequirements() {
+    requireEnvVar("GOOGLE_APPLICATION_CREDENTIALS");
+    requireEnvVar("GOOGLE_CLOUD_PROJECT");
+  }
+
+  @Before
+  public void setUp() throws InterruptedException, ExecutionException, IOException {
+    bout = new ByteArrayOutputStream();
+    out = new PrintStream(bout);
+    try (TranslationServiceClient client = TranslationServiceClient.create()) {
```

Comment:
A comment on complicated setUp's like these detailing what it is trying to accomplish would be nice.

Reply:
How would you feel if I just called the samples for these `setup` / `teardown` functions?

```
CreateGlossary.createGlossary(
        PROJECT_ID, "us-central1", GLOSSARY_ID, languageCodes, GLOSSARY_INPUT_URI);
```

Most of these are just doing the same thing as the samples, just without calling them directly. It'd be easier to maintain, but then relies on the samples themselves and I know some languages prefer not to do this. 

Create a glossary and then delete it afterwards.

Reply:
I'm +1 to relying on the samples.